engine/class_modules/apl/apl_death_knight.cpp

```diff
@@ -246,9 +246,9 @@ void frost( player_t* p )
   breath->add_action( "obliterate,target_if=max:(debuff.razorice.stack+1)%(debuff.razorice.remains+1)*death_knight.runeforge.razorice,if=buff.killing_machine.react&!variable.frostscythe_priority" );
   breath->add_action( "frostscythe,if=variable.frostscythe_priority&(buff.killing_machine.react|runic_power>45)" );
   breath->add_action( "obliterate,target_if=max:(debuff.razorice.stack+1)%(debuff.razorice.remains+1)*death_knight.runeforge.razorice,if=runic_power.deficit>40|buff.pillar_of_frost.up" );
-  breath->add_action( "death_and_decay,if=runic_power<36&rune.time_to_2>runic_power%18" );
   breath->add_action( "remorseless_winter,if=runic_power<36&rune.time_to_2>runic_power%18" );
-  breath->add_action( "howling_blast,if=runic_power<36&rune.time_to_2>runic_power%18" );
+  breath->add_action( "death_and_decay,if=talent.unholy_ground&!death_and_decay.ticking&runic_power.deficit>=10&!variable.2h_check|runic_power<36&rune.time_to_2>runic_power%18" );
+  breath->add_action( "howling_blast,if=!talent.unholy_ground|variable.2h_check&runic_power<36&rune.time_to_2>runic_power%18" );
```

Comment:
The 2h checks here should be removed. I copied your 2h test and removed the 2h check lines and saw a gain https://www.raidbots.com/simbot/report/6SkzUTYwyjdQqS1Cyrht79

The howling blast condition should have the check for unholy ground removed. When we are in the low runes, low rp state it will try to use death and decay first, but if unholy ground is talented and there are no available dnd charges then it will skip the howling blast line completely.